osgp/platform/osgp-adapter-domain-core/src/main/java/org/opensmartgridplatform/adapter/domain/core/application/services/FirmwareManagementService.java

```diff
@@ -153,27 +155,69 @@ private static String getFirmwareFilename(final String firmwareUrl) {
 
     public void handleSsldPendingFirmwareUpdate(final String deviceIdentification) {
 
-        final SsldPendingFirmwareUpdate ssldPendingFirmwareUpdate = this.ssldPendingFirmwareUpdateRepository
+        final List<SsldPendingFirmwareUpdate> ssldPendingFirmwareUpdates = this.ssldPendingFirmwareUpdateRepository
                 .findByDeviceIdentification(deviceIdentification);
 
-        if (ssldPendingFirmwareUpdate != null) {
-            final String organisationIdentification = ssldPendingFirmwareUpdate.getOrganisationIdentification();
-            final String correlationUid = ssldPendingFirmwareUpdate.getCorrelationUid();
+        if (CollectionUtils.isEmpty(ssldPendingFirmwareUpdates)) {
+            return;
+        }
+        /*
+         * A pending firmware update record was stored for this device earlier.
+         * This means this method is probably called following a firmware
+         * update. Retrieve the firmware version from the device to have the
+         * current version that is installed available.
+         *
+         * If multiple pending update records exist, it is not really clear what
+         * to do. The following approach assumes the most recently created one
+         * is relevant, and other pending records are out-dated.
+         */
+        final SsldPendingFirmwareUpdate ssldPendingFirmwareUpdate = this
```

Comment:
I am going to create a new item on the backlog for improving the firmware update process (including bringing the registered version up to date as soon as possible). I will include some information about the possibilities that may lay in using the available tables in another manner, instead of keeping track of a pending firmware update table.